### ISSUE TITLE
NuGet.Frameworks not found when referencing any local project/solution.

### ISSUE THREAD

**Issue by Makciek** [waf/CSharpRepl#246]
### Version

0.5.1

### What happened?

Hello
I've just started using CSharpRepl and wanted to reference my solution but unfortunately, that's constantly breaking. It's the same even with the newly created, clean, template project from VS.
Steps:
1. I'm starting repl by `csharprepl`
2. `> #r "path/to/.sln||.csproj"`
3. Build goes fine
4. 'Adding references from built project...' breaks
3.1 
![image](https://user-images.githubusercontent.com/15693079/233297159-785ae9c7-d4e1-44f1-93f5-fb2165099a7f.png)

What I tried to fix it:
1. Updated .net sdk
2. Went form VS 22 to VS 22 preview(so you can see the preview on the screenshot)
3. Repaired the VS
5. Cleaned the nuget caches
6. Added the missing lib to the project as a nugget reference

Have you seen such a problem before/know how to fix it?

**Comment by waf** [comment]
Hi, haven't seen this before. Is the source code of this solution available anywhere? 

If you can't share it, then just an excerpt of the CleanProject.csproj would help. Thanks!

**Comment by mars-low** [comment]
It can be reproduced in Docker container by running:
```shell
docker run -ti mcr.microsoft.com/dotnet/sdk
```
and then:
```shell
dotnet tool install -g csharprepl
export PATH="$PATH:/root/.dotnet/tools"
dotnet new console --name TestRepl
csharprepl --reference TestRepl/TestRepl.csproj
``` 

<details><summary><code>dotnet --info</code></summary>
<pre><code>.NET SDK:
 Version:   7.0.203
 Commit:    5b005c19f5
Runtime Environment:
 OS Name:     debian
 OS Version:  11
 OS Platform: Linux
 RID:         debian.11-x64
 Base Path:   /usr/share/dotnet/sdk/7.0.203/
Host:
  Version:      7.0.5
  Architecture: x64
  Commit:       8042d61b17
.NET SDKs installed:
  7.0.203 [/usr/share/dotnet/sdk]
.NET runtimes installed:
  Microsoft.AspNetCore.App 7.0.5 [/usr/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 7.0.5 [/usr/share/dotnet/shared/Microsoft.NETCore.App]
Other architectures found:
  None
Environment variables:
  Not set
global.json file:
  Not found
Learn more:
  https://aka.ms/dotnet/info
Download .NET:
  https://aka.ms/dotnet/download
</code></pre>
</details> 

<details><summary><code>csharprepl --reference TestRepl/TestRepl.csproj</code></summary>
<pre><code>Welcome to the C# REPL (Read Eval Print Loop)!
Type C# expressions and statements at the prompt and press Enter to evaluate them.
Type help to learn more, and type exit to quit.

Adding supplied references...
Building /TestRepl/TestRepl.csproj
MSBuild version 17.5.1+f6fdcf537 for .NET
  Determining projects to restore...
  All projects are up-to-date for restore.
  TestRepl -> /TestRepl/bin/Debug/net7.0/TestRepl.dll

Build succeeded.
    0 Warning(s)
    0 Error(s)

Time Elapsed 00:00:01.66
Adding references from built project...
Msbuild failed when processing the file '/TestRepl/TestRepl.csproj' with message: /usr/share/dotnet/sdk/7.0.203/Sdks/Microsoft.NET.Sdk/targets/Microsoft.NET.Sdk.Framework
ReferenceResolution.targets: (90, 5): The "ProcessFrameworkReferences" task failed unexpectedly.
System.IO.FileLoadException: Could not load file or assembly 'NuGet.Frameworks, Version=6.5.0.136, Culture=neutral, PublicKeyToken=31bf3856ad364e35'. Could not find or lo
ad a specific file. (0x80131621)
File name: 'NuGet.Frameworks, Version=6.5.0.136, Culture=neutral, PublicKeyToken=31bf3856ad364e35'
 ---> System.IO.FileLoadException: Could not load file or assembly 'NuGet.Frameworks, Version=6.5.0.136, Culture=neutral, PublicKeyToken=31bf3856ad364e35'.
   at System.Runtime.Loader.AssemblyLoadContext.<LoadFromPath>g____PInvoke|5_0(IntPtr ptrNativeAssemblyBinder, UInt16* ilPath, UInt16* niPath, ObjectHandleOnStack retAsse
mbly)
   at System.Runtime.Loader.AssemblyLoadContext.LoadFromAssemblyPath(String assemblyPath)
   at System.Runtime.Loader.AssemblyLoadContext.ResolveUsingLoad(AssemblyName assemblyName)
   at System.Runtime.Loader.AssemblyLoadContext.Resolve(IntPtr gchManagedAssemblyLoadContext, AssemblyName assemblyName)
   at Microsoft.NET.Build.Tasks.ProcessFrameworkReferences.<>c.<ExecuteCore>b__141_0(ITaskItem item)
   at System.Linq.Enumerable.SelectArrayIterator`2.MoveNext()
   at System.Linq.Enumerable.WhereEnumerableIterator`1.ToList()
   at System.Linq.Enumerable.ToList[TSource](IEnumerable`1 source)
   at Microsoft.NET.Build.Tasks.ProcessFrameworkReferences.ExecuteCore()
   at Microsoft.NET.Build.Tasks.TaskBase.Execute()
   at Microsoft.Build.BackEnd.TaskExecutionHost.Microsoft.Build.BackEnd.ITaskExecutionHost.Execute()
   at Microsoft.Build.BackEnd.TaskBuilder.ExecuteInstantiatedTask(ITaskExecutionHost taskExecutionHost, TaskLoggingContext taskLoggingContext, TaskHost taskHost, ItemBuck
et bucket, TaskExecutionMode howToExecuteTask)
</code></pre>
</details> 

Besides failed task, project seems to be imported correctly and doesn't cause additional problems.

**Comment by waf** [comment]
Some possibly related issues:

https://github.com/stride3d/stride/issues/1453#issuecomment-1433729780
https://github.com/microsoft/MSBuildLocator/issues/86#issuecomment-759183289

**Comment by waf** [comment]
This is now available in v0.6 -- run `dotnet tool update --global csharprepl` to update to the newest version. Thanks again!

**Comment by Makciek** [comment]
That fixed the issue for me, thanks!
